test/search_api/test_models.py

```diff
@@ -12,123 +11,123 @@
 }
 
 DATASET_ICAT_DATA = {
-    "endDate": "2000-12-31 00:00:00+00:00",
+    "endDate": "2000-12-31T00:00:00.000Z",
```

Comment:
Does Python ICAT returns `datetime` values in this format? The tests in this module convert data from ICAT to PaNOSC format so the `datetime` values in the ICAT data dictionaries should be in the correct format. Not sure if the old format is the right one as I cannot remember whether I looked this up when I wrote the tests so best to double check.

Reply:
The original format is the Python ICAT format, I changed it because the `DateHandler` conversion functions (str to datetime I think?) didn't seem to work. So in an attempt to just get something to work, I just changed all the datetime string values to one that the conversion actually can convert. To move back to the original, `DateHandler` will have to be changed so it can convert that format. That would be the 'best practice' solution, but I'm not sure how time consuming it'll be to change it.

Reply:
I see why you changed it. If you use `DateHandler.str_to_datetime_object`, which I initially used, then that only converts a `string` to a `datetime` object whereas the updated tests now expect the actual `datetime` values to be of type `string`. To get it to work I think that we just need to convert the `datetime` object generated by `DateHandler.str_to_datetime_object` to a `string` but in a Search API format. Doing this would allow us to change the `datetime` values back to the original format.

If we take the `creationDate` key in the `DATASET_PANOSC_DATA` dictionary as an example, I think that we can get get it to work like this:
```python
"creationDate": DateHandler.str_to_datetime_object(DATASET_ICAT_DATA["createTime"]).strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
```

or by importing `SearchAPIDatetime` from `models.py` and doing this if we want to keep the `datetime` `string` format in one place:
```python
"creationDate": SearchAPIDatetime.use_search_api_format(DateHandler.str_to_datetime_object(DATASET_ICAT_DATA["createTime"]))
```

Reply:
I removed https://github.com/ral-facilities/datagateway-api/commit/6242fe1e9ede39ed696feb567c4238e29dde6eb6 and made the change using `use_search_api_format()`. On data which is in the PaNOSC data model (e.g. the changes in `test_search_endpoint.py`), I've searched and replaced those

Reply:
Thanks for sorting this @MRichards99, glad it worked. Looks good, approving now.